packages/es-components/src/components/navigation/useNavigation.tsx

```diff
@@ -23,32 +29,32 @@ const HorizontalNavigation = styled.nav`
   }
 `;
 
-export function useNavigation(orientation) {
+export function useNavigation(orientation: string) {
   const OrientedNavigation =
     orientation === 'horizontal' ? HorizontalNavigation : VerticalNavigation;
 
-  function Navigation(props) {
+  function Navigation(props: NavigationProps) {
```

Comment:
Defining a Component within a `use` call is an anti-pattern. We should take a look at fixing this.